backtracepython/attributes/attribute_manager.py

```diff
@@ -1,51 +1,44 @@
 import platform
 
-from backtracepython.attributes.backtrace_attribute_provider import (
-    BacktraceAttributeProvider,
-)
-from backtracepython.attributes.linux_memory_attribute_provider import (
-    LinuxMemoryAttributeProvider,
-)
-from backtracepython.attributes.machine_attribute_provider import (
-    MachineAttributeProvider,
-)
-from backtracepython.attributes.machineId_attribute_provider import (
-    MachineIdAttributeProvider,
-)
-from backtracepython.attributes.process_attribute_provider import (
-    ProcessAttributeProvider,
-)
-from backtracepython.attributes.session_attribute_provider import (
-    SessionAttributeProvider,
-)
-from backtracepython.attributes.system_attribute_provider import SystemAttributeProvider
+from .backtrace_attribute_provider import BacktraceAttributeProvider
+from .linux_memory_attribute_provider import LinuxMemoryAttributeProvider
+from .machine_attribute_provider import MachineAttributeProvider
+from .machineId_attribute_provider import MachineIdAttributeProvider
+from .process_attribute_provider import ProcessAttributeProvider
+from .report_data_builder import ReportDataBuilder
+from .session_attribute_provider import SessionAttributeProvider
+from .system_attribute_provider import SystemAttributeProvider
+from .user_attribute_provider import UserAttributeProvider
 
 
 class AttributeManager:
     def __init__(self):
-        self.dynamic_attributes = self.get_predefined_dynamic_attribute_providers()
-        self.scoped_attributes = {}
-        for (
-            scoped_attribute_provider
-        ) in self.get_predefined_scoped_attribute_providers():
-            self.try_add(self.scoped_attributes, scoped_attribute_provider)
+        self.attribute_providers = (
+            self.get_predefined_dynamic_attribute_providers()
+            + self.get_predefined_scoped_attribute_providers()
+        )
 
     def get(self):
-        result = {}
-        for dynamic_attribute_provider in self.dynamic_attributes:
-            self.try_add(result, dynamic_attribute_provider)
-        result.update(self.scoped_attributes)
+        attributes = {}
+        annotations = {}
+        for attribute_provider in self.attribute_providers:
 
-        return result
+            generated_attributes, generated_annotations = ReportDataBuilder.get(
+                self.try_get(attribute_provider)
+            )
+            attributes.update(generated_attributes)
+            annotations.update(generated_annotations)
+
+        return attributes, annotations
 
     def add(self, attributes):
-        self.scoped_attributes.update(attributes)
+        self.attribute_providers.append(UserAttributeProvider(attributes))
 
-    def try_add(self, dictionary, provider):
+    def try_get(self, provider):
         try:
-            dictionary.update(provider.get())
+            return provider.get()
         except:
-            return
+            return {}
```

Comment:
why not just `try/catch` in the `get` method itself, and `continue` if `get` throws?